scripts/polarity-diff-report

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+
+import argparse
+import csv
+import json
+import os
+import sys
+import time
+from json.decoder import JSONDecodeError
+
+import ctakesclient
+from ctakesclient.typesystem import MatchText, Polarity
+
+
+def default(parser, args):
+    del args
+
+    parser.print_help()
+
+
+def compare_note_polarity(text: str):
+    """
+    Prints difference summary (if any)
+    """
+    ner = ctakesclient.client.extract(text)
+
+    matches = ner.list_match()
+    spans = ner.list_spans(matches)
+
+    polarities_cnlp = ctakesclient.transformer.list_polarity(text, spans)
+    if len(matches) != len(polarities_cnlp):
+        raise JSONDecodeError('Polarity lists had different lengths!')
+
+    differing = []
+    for i in range(len(matches)):
+        if matches[i].polarity.value != polarities_cnlp[i].value:
+            differing.append({
+                'cnlp_polarity': polarities_cnlp[i].value,
+                'match': matches[i].as_json(),
+            })
+
+    return differing
+
+
+def calculate(parser, args):
+    del parser
+
+    output_path = os.path.splitext(args.notes_path)[0] + '.pdr.ndjson'
+    if os.path.exists(output_path) and not args.resume:
+        print('Output file already exists. Use --continue if this is intended.',
+              file=sys.stderr)
+        sys.exit(1)
+
+    print("Processing...")
+
+    if args.resume:
+        mode = 'r+'
+    else:
+        mode = 'w'
+
+    total_tic = time.perf_counter()
+    with open(output_path, mode, buffering=1, encoding='utf8') as output_file:
+        start_processing_after = None
+        if args.resume:
+            # Read until the end
+            for line in output_file:
+                pass
+            if line:
+                start_processing_after = json.loads(line)['instance_num']
+                print(f"Resuming after instance id {start_processing_after}...")
+
+        with open(args.notes_path, 'r', encoding='utf8') as notes_file:
+            notes = csv.DictReader(notes_file)
+            for count, note in enumerate(notes, start=1):
+                if start_processing_after is not None:
+                    if note['INSTANCE_NUM'] == start_processing_after:
+                        start_processing_after = None
+                    continue
+
+                note_tic = time.perf_counter()
+                try:
+                    error = None
+                    differences = compare_note_polarity(note['OBSERVATION_BLOB'])
+                except JSONDecodeError as e:
+                    error = str(e)
+                    differences = []
+
+                json.dump({
+                    # Save the instance id for later sanity checking
+                    'instance_num': note['INSTANCE_NUM'],
```

Comment:
It appears that `INSTANCE_NUM` is unique among all entries in the data source file I have. Is that correct that it's basically a unique id per row (note/encounter/patient/etc)?